cairo/src/runner.rs

```diff
@@ -0,0 +1,472 @@
+//! This module represents a run of a Cairo program as a consecution of execution
+//! steps, each of which define the execution logic of Cairo instructions
+
+use crate::definitions::*;
+use crate::memory::CairoMemory;
+use crate::word::{CairoWord, Decomposition};
+use ark_ff::FftField;
+
+/// A structure to store program counter, allocation pointer and frame pointer
+#[derive(Clone, Copy)]
+pub struct CairoPointers<F: FftField> {
+    /// Program counter: points to address in memory
+    pub pc: F,
+    /// Allocation pointer: points to first free space in memory
+    pub ap: F,
+    /// Frame pointer: points to the beginning of the stack in memory (for arguments)
+    pub fp: F,
```

Comment:
should these really be public? My understanding is that you should not set them directly, and they should be set only through the state transition

Reply:
Absolutely right. I should have a getter for these, as I really only extract them. One shouldn't be able to tamper them.

Reply:
Actually I think I don't even need a getter, as I am only using them within the same module